### ISSUE TITLE
xz package does not provide unversioned liblzma.so symlink

### ISSUE THREAD

**Issue by jkoritzinsky** [microsoft/azurelinux#9217]
**Describe the bug**
The xz package provides versioned SOs, `/usr/lib/liblzma.so.5` and `/usr/lib/liblzma.5.4.4`, but it does not provide `/usr/lib/liblzma.so`. This breaks any tooling that looks up liblzma.so without a version, such as through autotools.

This was found on a docker image based on azurelinuxpreview.azurecr.io/public/azurelinux/base/core:3.0 with no additional packages installed.

**To Reproduce**
1. Run `docker run --rm -it azurelinuxpreview.azurecr.io/public/azurelinux/base/core:3.0 bash`
2. In the container, run `ls /usr/lib/liblzma.so*`

**Expected behavior**

There should be three files, `liblzma.so`, `liblzma.so.5` and `liblzma.so.5.4.4`, where they each symlink (possibly transitively) to `liblzma.so.5.4.4`

**Comment by zcobol** [comment]
@jkoritzinsky if you inspect the `xz.spec` the link is created during package installation:

`ln -svf "../..%{_libdir}/$(readlink %{buildroot}%{_libdir}/liblzma.so)" %{buildroot}%{_libdir}/liblzma.so`

and `ls` output:

```
zcobol@azurelinux [ ~ ]$ ls -l /usr/lib/liblzma.so
lrwxrwxrwx 1 root root 30 Apr 23  2022 /usr/lib/liblzma.so -> ../../usr/lib/liblzma.so.5.2.5
```

I suspect it's a Docker container assembly issue.

**Comment by jkoritzinsky** [comment]
Yes, I suspect that it is an issue with the Azure Linux 3.0 docker image construction (which I believe is also defined in this repo).

**Comment by jkoritzinsky** [comment]
Looping back to this, it's still an issue with the released AzureLinux 3.0 base image. It wasn't a one-off image build issue.

**Comment by eric-desrochers** [comment]
Look like you will need `xz-devel` which provides ` /usr/lib/liblzma.so`

xz-devel-5.4.4-1.azl3.x86_64 : Header and development files for xz
Repo        : azurelinux-official-base
Matched from:
Filename    : /usr/lib/liblzma.so
